gtfu/src/main/java/gtfu/tools/DayLogSlicer.java

```diff
@@ -59,15 +63,25 @@ public DayLogSlicer(TripCollection tripCollection, RouteCollection routeCollecti
             agentMap.put(tripID,agent);
 
             String latLon = String.valueOf(lat) + String.valueOf(lon);
+
             // If there is no LatLonMap for this trip, add it.
             if(gpsMap.get(tripID) == null){
                 Map<String, GPSData> latLonMap = new HashMap();
                 gpsMap.put(tripID,latLonMap);
             }
-            // If there is no existing GPSData for this latLon value, add it
+
+            // If there is no existing GPSData for this latLon value, add it and update previousUpdate for that trip
             if (gpsMap.get(tripID).get(latLon) == null) {
-                gpsMap.get(tripID).put(latLon, new GPSData(seconds * 1000l, lat, lon));
+                // previousUpdateMap stores the most recent GPS update timestamp (in seconds) for each trip_id.
+                // it relies on the list being sorted by timestmap, which it is.
+                int secsSinceLastUpdate = -1;
+                if (previousUpdateMap.get(tripID) != null) {
+                    secsSinceLastUpdate = seconds - previousUpdateMap.get(tripID);
+                }
+                gpsMap.get(tripID).put(latLon, new GPSData(seconds * 1000l, secsSinceLastUpdate, lat, lon));
```

Comment:
The expression  `gpsMap.get(tripID)` appears at least four times in the code above. Most modern compilers have Common Subexpression Elimination, so likely not a performance issue, but still overly verbose and a potential cause of errors. Just assign to a variable up top and then reference that